src/analyses/base.ml

```diff
@@ -462,7 +462,7 @@ struct
         let var = get_var a gs st x in
         let v = VD.eval_offset (Queries.to_value_domain_ask a) (fun x -> get a gs st x exp) var offs exp (Some (Var x, Offs.to_cil_offset offs)) x.vtype in
         if M.tracing then M.tracec "get" "var = %a, %a = %a\n" VD.pretty var AD.pretty (AD.of_mval (x, offs)) VD.pretty v;
-        if full then v else match v with
+        if full then var else match v with
           | Blob (c,s,_) -> c
           | x -> x
```

Comment:
It's not clear to me whether this is the right thing. It looks to me that the `~full:true` argument is just supposed to avoid the usual `Blob` unwrapping because `BlobSize` needs to take the size from there.
Using `var` instead of `v` means that offsets are actually ignored which seems odd. Is that really the right thing in general?

Reply:
I thought about this a bit more and maybe have some answer to my own question.
The variable and its offsets here are already resolved mvalues in an address set, e.g. `x.foo` or `(alloc@...).bar`. Offsets within the blob don't matter for the size of the blob, so I suppose it's safe to ignore them.

In this light, it becomes questionable what a `BlobSize (alloc@...).bar` should even be. The least error prone thing would be to forbid offsets in the query altogether (if not by crashing, then by returning `bot`). Similarly, if the variable isn't a heap variable, then the same thing could apply.

Reply:
>  The least error prone thing would be to forbid offsets in the query altogether (if not by crashing, then by returning bot). Similarly, if the variable isn't a heap variable, then the same thing could apply.

If we do this, I think it'll be good to merge after!

Reply:
> > The least error prone thing would be to forbid offsets in the query altogether (if not by crashing, then by returning bot). Similarly, if the variable isn't a heap variable, then the same thing could apply.
> 
> If we do this, I think it'll be good to merge after!

I added some changes in https://github.com/goblint/analyzer/pull/1125/commits/15a5b1a1470014017df01ba51e1ddf41c35a157f which should hopefully reflect what was suggested in the discussion here.

`make test` runs through successfully, so hopefully the changes make sense